connectors/src/connectors/zendesk/lib/data_cleanup.ts

```diff
@@ -0,0 +1,128 @@
+  ZendeskArticleResource,
+  ZendeskBrandResource,
+  ZendeskCategoryResource,
+  ZendeskTicketResource,
+} from "@connectors/resources/zendesk_resources";
+import type { DataSourceConfig } from "@connectors/types/data_source_config";
+
+/**
+ * Deletes all the data relative to a brand (tickets, help center, categories, articles) in the data source and in the db.
+ */
+export async function deleteBrand({
+  connectorId,
+  brandId,
+  dataSourceConfig,
+}: {
+  connectorId: number;
+  brandId: number;
+  dataSourceConfig: DataSourceConfig;
+}) {
+  await Promise.all([
+    deleteBrandHelpCenter({ connectorId, brandId, dataSourceConfig }),
+    deleteBrandTickets({ connectorId, brandId, dataSourceConfig }),
+  ]);
+  await ZendeskBrandResource.deleteByConnectorId(connectorId);
+}
+
+/**
+ * Deletes all the tickets stored in the db and in the data source relative to a brand.
+ */
+export async function deleteBrandTickets({
+  connectorId,
+  brandId,
+  dataSourceConfig,
+}: {
+  connectorId: number;
+  brandId: number;
+  dataSourceConfig: DataSourceConfig;
+}) {
+  const tickets = await ZendeskTicketResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  /// deleting the tickets in the data source
+  await Promise.all(
+    tickets.map((ticket) =>
+      deleteFromDataSource(
+        dataSourceConfig,
+        getTicketInternalId(ticket.connectorId, ticket.ticketId)
+      )
+    )
+  );
+  /// deleting the tickets stored in the db
+  await ZendeskTicketResource.deleteByBrandId({ connectorId, brandId });
+}
+
+/**
+ * Deletes all the data stored in the db and in the data source relative to a brand's help center (category, articles).
+ */
+export async function deleteBrandHelpCenter({
+  connectorId,
+  brandId,
+  dataSourceConfig,
+}: {
+  connectorId: number;
+  brandId: number;
+  dataSourceConfig: DataSourceConfig;
+}) {
+  /// deleting the articles in the data source
+  const articles = await ZendeskArticleResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  await Promise.all(
+    articles.map((article) =>
+      deleteFromDataSource(
+        dataSourceConfig,
+        getArticleInternalId(connectorId, article.articleId)
+      )
+    )
+  );
+  /// deleting the articles stored in the db
+  await ZendeskArticleResource.deleteByBrandId({
+    connectorId,
+    brandId,
+  });
+  /// deleting the categories stored in the db
+  await ZendeskCategoryResource.deleteByBrandId({ connectorId, brandId });
+}
+
+/**
+ * Deletes all the data stored in the db and in the data source relative to a category (articles).
+ */
+export async function deleteCategory({
+  connectorId,
+  categoryId,
+  dataSourceConfig,
+}: {
+  connectorId: number;
+  categoryId: number;
+  dataSourceConfig: DataSourceConfig;
+}) {
+  /// deleting the articles in the data source
+  const articles = await ZendeskArticleResource.fetchByCategoryId({
+    connectorId,
+    categoryId,
+  });
+  await Promise.all(
+    articles.map((article) =>
```

Comment:
Same here, how many articles can this be?

Reply:
based on current customer cases, this ranges from 10 to 50 articles typically, but this could be more for some other customer in the future.
I'm thinking about batching this in a next PR (there are a few things to move around) to avoid making this one grow even more.